### PR TITLE
Polyfill - places react-app-polyfill everywhere that generates apps inside UI Fabric repo

### PR DESCRIPTION
**Pull request by kenotron**
#### Pull request checklist

- [x] Include a change request file using `$ npm run change`

#### Description of changes

Adds a react-app-polyfill dependency / devDependency in places where apps are built in Fabric (npm start, npm run build) 

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9622)

### CODE REVIEW

**Comment by dzearing** [comment]
Looks good. Thank you for fixing!

File: scripts/webpack/webpack-resources.js [hunk 1]
```diff
@@ -1,11 +1,55 @@
...
+  }
+}
+
+/**
+ * Prepends the entry points with a react 16 compatible polyfill but only for sites that have react as a dependency
+ */
+function prependEntryWithPolyfill(entry) {
+  const packageJson = JSON.parse(fs.readFileSync(path.join(process.cwd(), 'package.json'), 'utf-8'));
+  if (
+    (packageJson.dependencies && Object.keys(packageJson.dependencies).includes('react')) ||
```
Comment (ecraig12345) [inline scripts/webpack/webpack-resources.js]: What about peer deps?

Comment (kenotron) [reply scripts/webpack/webpack-resources.js]: I shouldnt dictate the polyfill for our library consumers. I think we should leave the peer deps alone..

**Comment by kenotron** [comment]
The size increase is expected as that's the point of the pr - to bundle in the polyfill.

**Comment by JasonGore** [comment]
Did you check these perf test numbers? I expect 1-3 to be off, and in varying directions, but a majority of these are showing an uptick, which seems odd.

**Comment by JasonGore** [comment]
This PR definitely seems to be triggering differing versions of functions which are more expensive. Here BaseButton, for example, "before" is on top. Additional V8 functions are being called after that seem to be more expensive for the same functionality.

image

In general, it seems that `assign` is 2x more expensive, which adds up for more complex components:

image

This seems to be the case all over, and I would say this PR has cost us a 10%+ perf hit, based on these results.

**Comment by kenotron** [comment]
This is just a liberal use of polyfill across all our apps. Perf-test is a special case where i KNOW is using puppeteer + Chromium, so I"m going to special case and not apply polyfill there.

**Pull request merged**